File: src/main/java/edu/berkeley/bid/bullet/BodyInfo.java
package edu.berkeley.bid.bullet;

import java.io.Serializable;

public class BodyInfo implements Serializable {
    private static final long serialVersionUID = 8947527343235449L;

    public String m_baseName;
    public String m_bodyName; 

    public BodyInfo (String baseName, String bodyName) {
	m_baseName = baseName;
	m_bodyName = bodyName;
    }

    public BodyInfo() {
    }

    @Override
    public String toString () {
	return "BodyInfo[baseName=" + m_baseName + ",bodyName=" + m_bodyName + "]";
    }

}

File: src/main/scala/BIDMach/rl/environments/bullet/BodyInfo.scala
package BIDMach.rl.environments.bullet;

@SerialVersionUID(473252348962354L)
class BodyInfo(jbi:edu.berkeley.bid.bullet.BodyInfo) extends Serializable {
    val javaBodyInfo:edu.berkeley.bid.bullet.BodyInfo = jbi;

    def baseName:String = javaBodyInfo.m_baseName;
    def bodyName:String = javaBodyInfo.m_bodyName;

    def baseName_=(s:String) = javaBodyInfo.m_baseName = s;
    def bodyName_=(s:String) = javaBodyInfo.m_bodyName = s;

    def this() = this(new edu.berkeley.bid.bullet.BodyInfo());
	
    override def toString () = {
	"BodyInfo[baseName=" + baseName + ",bodyName=" + bodyName + "]";
    }

}
